src/fetch.rs

```diff
@@ -0,0 +1,104 @@
+// Tako -- Take container image.
+// Copyright 2018 Arian van Putten, Ruud van Asseldonk, Tako Marks.
+
+//! Contains the main fetching logic (downloading manifests and images).
+
+use std::fs;
+use std::io;
+use std::io::{BufRead, Read, Write};
+
+use config::Config;
+use curl;
+use error::{Error, Result};
+use manifest::Manifest;
+
+fn load_config(config_fname: &str) -> Result<Config> {
+    let f = fs::File::open(config_fname)?;
+    let buf_reader = io::BufReader::new(f);
+    let mut lines = Vec::new();
```

Comment:
Why wouldn't   `Config::parse(buf_reader.lines())` work here?  Both are iterators. Why convert to a vector first, instead of streaming to the parser immediately?

Reply:
`buf_reader.lines()` returns an `Iterator<Item = Result<String, io::Error>>`, but `Config::parse` expects an `Iterator<Item = String>`. We could do `.lines().map(|x| x.unwrap())`, but then we don't get proper error handling in the error case. Ideally we would be able to write `.lines().map(|x| x?)`, but that does not work unfortunately:

    error[E0277]: the `?` operator can only be used in a function that returns `Result`

It would make the closure return a `Result`, rather than making the outer function return.

Now that I am looking into this, there is [`from_iter`](https://doc.rust-lang.org/std/result/enum.Result.html#impl-FromIterator%3CResult%3CA%2C%20E%3E%3E) and [`collect`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect) which do make this a bit nicer.

Fixed in fbb0ac6e6b8a28e60266c0152926333dbc1a243e.